src/Marketplace/Client.php

```diff
@@ -47,26 +54,51 @@ public function __construct()
      */
     public function get($endpoint, $params = [])
     {
-        $endpoint = collect([$this->domain, self::API_PREFIX, $endpoint])->implode('/');
+        $lock = $this->lock(static::LOCK_KEY, 10);
+
+        try {
+            $lock->block(5);
+
+            $endpoint = collect([$this->domain, self::API_PREFIX, $endpoint])->implode('/');
 
-        $key = 'marketplace-'.md5($endpoint.json_encode($params));
+            $key = 'marketplace-'.md5($endpoint.json_encode($params));
 
-        return Cache::rememberWithExpiration($key, function () use ($endpoint, $params) {
-            $response = Guzzle::request('GET', $endpoint, [
-                'verify' => $this->verifySsl,
-                'query' => $params,
-            ]);
+            return $this->cache()->rememberWithExpiration($key, function () use ($endpoint, $params) {
+                $response = Guzzle::request('GET', $endpoint, [
+                    'verify' => $this->verifySsl,
+                    'query' => $params,
+                ]);
 
-            $json = json_decode($response->getBody(), true);
+                $json = json_decode($response->getBody(), true);
 
-            return [$this->cache => $json];
-        });
+                return [60 => $json];
+            });
+        } catch (LockTimeoutException $e) {
+            return $this->cache()->get($key);
+        } finally {
+            $lock->release();
+        }
     }
 
-    public function cache($cache = 60)
+    private function cache()
```

Comment:
It doesn't look like we're actually calling this `cache` method anywhere, and since it's not documented, I thought it'd be okay to repurpose.